src/main/java/core/basesyntax/strategy/impl/BalanceOperationHandler.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.strategy.mapvalidator.KeyAbsentValidator;
+import java.util.Map;
+
+public class BalanceOperationHandler
+        extends AbstractOperationHandler
+        implements KeyAbsentValidator {
+    public BalanceOperationHandler() {
+        super(Operation.BALANCE);
+    }
+
+    @Override
+    protected void processTransaction(Map<String, Integer> fruitQuantityMap,
+                                      FruitTransaction transaction) {
+        validateMap(fruitQuantityMap, transaction);
```

Comment:
The `validateMap` method call suggests that you are doing some validation on the `fruitQuantityMap` and `transaction`. However, according to the checklist, we should avoid having logic other than the operation's logic in the handler. Ensure that the validation logic is decoupled and does not violate the SRP.